### PR TITLE
Revert "ci: set_assignee: pick next area when submitter = assignee"

### PR DESCRIPTION
**Pull request by fabiobaltieri**
This reverts commit 913426e585e9dc72b9bc3eb04f44be192ccbfd6a, to be causing PRs from being misassigned in a confusing way, causing extra work to the misassigned assignee to reassign manually, for example:

- https://github.com/zephyrproject-rtos/zephyr/pull/77938
- https://github.com/zephyrproject-rtos/zephyr/pull/77803
- https://github.com/zephyrproject-rtos/zephyr/pull/77876

### CODE REVIEW

**Review by henrikbrixandersen — APPROVED** [review]
Thanks. I've been meaning to suggest the same.

**Review by danieldegrasse — APPROVED** [review]

**Review by nashif — CHANGES_REQUESTED** [review]
isnt the script doing what it is supposed to do? see https://github.com/zephyrproject-rtos/zephyr/pull/77876#issuecomment-2329057402

I would rather improve instead of completely revert, there are always going to be special cases, but the issues this prevents outweighs the false positives IMO

**Comment by fabiobaltieri** [comment]
> isnt the script doing what it is supposed to do? see [#77876 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/77876#issuecomment-2329057402)

Assigning PRs to other random maintainers that get confused and have to reassign it manually? Yeah I guess it is. The three I linked just happened to have flown under my radar yesterday.

**Comment by kartben** [comment]
> > isnt the script doing what it is supposed to do? see [#77876 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/77876#issuecomment-2329057402)
> 
> Assigning PRs to other random maintainers that get confused and have to reassign it manually? Yeah I guess it is. The three I linked just happened to have flown under my radar yesterday.

Are you suggesting to leave the PRs falling in the category "submitter = assignee for they are the maintainer" unassigned then?

**Comment by nashif** [comment]
> Assigning PRs to other random maintainers that get confused and have to reassign it manually? Yeah I guess it is. The three I linked just happened to have flown under my radar yesterday.

the code is "mostly" doing what it is supposed to do.

Maintainer A submits a change with most files from the area they maintain, but also other areas, say area B. Maintainer of area B (Maintainer B)  is assigned. 
A can't approve anyways, so, to make sure changes to area B do not go un-noticed, we ask Maintainer B to be the assignee and approve, so that the change does not go in without their approval. 

Maintainer B can approve and re-assign to Maintainer A if they feel the change is fine and A should be assigned.

as I said, there is some minor issue in the code that needs to be fixed, other than that, it is doing its thing.

**Comment by henrikbrixandersen** [comment]
> Are you suggesting to leave the PRs falling in the category "submitter = assignee for they are the maintainer" unassigned then?

PRs submitted by a single maintainer only touching the area of that maintainer is assigned to that maintainer regardless of this script.

**Comment by fabiobaltieri** [comment]
> A can't approve anyways, so, to make sure changes to area B do not go un-noticed, we ask Maintainer B to be the assignee and approve, so that the change does not go in without their approval.

Maintainer B (and C and E and the whole rest of the alphabet) are already assigned as reviewer and notified of the change, if they don't review the change then tough luck.

**Comment by nashif** [comment]
> Maintainer B (and C and E and the whole rest of the alphabet) are already assigned as reviewer and notified of the change, if they don't review the change then it's a different problem.

yes, it is a different problem when they start reverting PRs they were added to as reviewers but never got to review the PR for whatever reason.

It is also a problem when I touch 500 files related to my area and sneak in a kernel API change in 1 file, I get assigned to the PR and if nobody notices, i get 2 reviews from 2 random people  and it is a done deal and the code is in the tree. 

It is not a problem however to get assignees wrong, we have this all the time, it is expected and part of the responsibilities of assignees to re-assign.

**Comment by fabiobaltieri** [comment]
Sure, my point here is that from what I've seen when this kicks in it is going to cause confusion and manual work, and that is going to happen on many "normal" PR that do not fall in the scenario you described. All for the very slim chance that it's going to randomly assign the right person in PR that touch multiple areas.

I think we were better off with the previous behavior, and relying on the release team to catch the problematic cases you described. Anyway, happy to hear other people opinions at this point.

**Comment by MaureenHelm** [comment]
Another example: #75472

**Comment by nashif** [comment]
> Another example: #75472

 ./scripts/get_maintainer.py path drivers/sensor/espressif/esp32_temp/esp32_temp.c
Drivers: Sensors
	status: maintained
	maintainers: MaureenHelm
	collaborators: avisconti, teburd, yperess, tristan-google, ubieda, jeppenodgaard
	inform:
	labels: area: Sensors
	tests: drivers.sensors
	tags:
	description:


This should be included in the ESP area, then we will not have this. 

Right now this PR is changing multiple areas, submitter is assignee, so the rule applies: assign maintainer of other areas being touched.

**Comment by nashif** [comment]
> Another example: #75472

actually, the same assignment would have happened with this commit being reverted. This is the original way this script was working.

with the commit reverted...
>> candidate maintainers: {'sylvioalves': 18, 'MaureenHelm': 2, 'gmarull': 1, 'anangl': 0, 'nordic-krch': 0, 'teburd': 0, 'de-nordic': 0, 'dcpleung': 0, 'tbursztyka': 0, 'jukkar': 0}
Picked assignees: ['MaureenHelm'] (9.09% ownership)

**Comment by henrikbrixandersen** [comment]
> same thing with #77938, #77876 and #77803

That's an I interesting data point. I will withdraw my approval based on this.

**Comment by fabiobaltieri** [comment]
> same thing with #77938, #77876 and #77803

Yes because since d9a300e4ae29df48dc67ff648203066e3c5ab1d7 there's a weird logic that, if I'm reading it correctly, picks either:
- the non platforms area with the most changes (the `break case`) or
- the platform area with the least amount of changes (?!)

```py
    for area, count in area_counter.items():
        print(area, count, area.maintainers)
        if count == 0:
            continue
        if len(area.maintainers) > 0:
            assignees = area.maintainers

            if 'Platform' not in area.name:
                break
```

```console
$ ./scripts/set_assignees.py --dry-run -v  -P 77938
...
<Area ADI Platforms> 3 ['MaureenHelm']
<Area Google Platforms> 2 ['fabiobaltieri', 'keith-zephyr']
<Area Disk> 1 ['danieldegrasse']
Picked assignees: ['danieldegrasse'] (0.54% ownership)
```

```console
$ ./scripts/set_assignees.py --dry-run -v  -P 77876
<Area STM32 Platforms> 122 ['erwango']
<Area ADI Platforms> 3 ['MaureenHelm']
<Area Google Platforms> 2 ['fabiobaltieri', 'keith-zephyr']
<Area Drivers: Clock control> 0 ['nordic-krch']
Picked assignees: ['fabiobaltieri', 'keith-zephyr'] (1.14% ownership)
```

```console
$ ./scripts/set_assignees.py --dry-run -v  -P 75472
<Area Espressif Platforms> 18 ['sylvioalves']
<Area Drivers: Sensors> 2 ['MaureenHelm']
Picked assignees: ['MaureenHelm'] (9.09% ownership)
```

that's a separate issue

**Comment by nashif** [comment]
> * the platform area with the least amount of changes (?!)

that is a bug that I already mentioned above, i am working on a fix.

**Pull request closed without merging**
